### PR TITLE
fix: always use the regular function to avoid losing module context

### PR DESCRIPTION
**Pull request by latin-1**
Using an arrow function here would cause the loss of context (`this`) inside the module factory, which may break some modules that rely on it.

### CODE REVIEW

**Comment by pmmmwh** [comment]
Hey, thanks for the PR. I'll review this tomorrow and also investigate a bit into the issue.

Is there concrete example demonstrating how arrow functions would break?

**Comment by latin-1** [comment]
Assume we have a module, x.js:

```js
this.foo = "bar";
```

And now, we are trying to require it from node:

```sh
$ echo 'this.foo = "bar";' > x.js
$ node -e 'require("./x").foo' -p
bar
```

Webpack has a similar approach to Node.js by calling the module factory with `this = module.exports`. However, the react-refresh-webpack-plugin's runtime code breaks this treatment.

Currently, we are running into this bug with `@mediapipe/face_mesh`, which uses the same way to export something in both browsers and webpack.

File: lib/utils/makeRefreshRuntimeModule.js [hunk 1]
```diff
@@ -27,45 +27,44 @@ function makeRefreshRuntimeModule(webpack) {
...
-            `options.factory = ${runtimeTemplate.basicFunction(
-              'moduleObject, moduleExports, webpackRequire',
-              [
-                `${refreshGlobal}.setup(options.id);`,
-                'try {',
-                webpack.Template.indent(
-                  'originalFactory.call(this, moduleObject, moduleExports, webpackRequire);'
-                ),
-                '} finally {',
+            `options.factory = function(moduleObject, moduleExports, webpackRequire) {`,
```
Comment (pmmmwh) [inline lib/utils/makeRefreshRuntimeModule.js:30]:
```suggestion
            `options.factory = function (moduleObject, moduleExports, webpackRequire) {`,
```

File: lib/utils/makeRefreshRuntimeModule.js [hunk 2]
```diff
@@ -27,45 +27,44 @@ function makeRefreshRuntimeModule(webpack) {
...
                 ]),
+                '} else {',
+                webpack.Template.indent(`${refreshGlobal}.cleanup(options.id)`),
                 '}',
-              ]
-            )}`,
+              ]),
+              '}',
+            ]),
+            `}`,
```
Comment (pmmmwh) [inline lib/utils/makeRefreshRuntimeModule.js:67]:
```suggestion
            `};`,
```

File: test/unit/makeRefreshRuntimeModule.test.js [hunk 3]
```diff
@@ -119,7 +119,7 @@ __webpack_require__.$Refresh$ = {
     expect(runtime).toMatchInlineSnapshot(`
 "__webpack_require__.i.push((options) => {
 	const originalFactory = options.factory;
-	options.factory = (moduleObject, moduleExports, webpackRequire) => {
+	options.factory = function(moduleObject, moduleExports, webpackRequire) {
```
Comment (pmmmwh) [inline test/unit/makeRefreshRuntimeModule.test.js:122]:
```suggestion
	options.factory = function (moduleObject, moduleExports, webpackRequire) {
```

**Review by pmmmwh — APPROVED** [review]

**Pull request merged**